src/devices/Arduino/SupportedMode.cs

```diff
@@ -1,95 +1,158 @@
 
         /// <summary>
         /// The pin can perform UART (TX or RX)
         /// </summary>
-        Serial = (0x0A),
+        public static SupportedMode Serial = new SupportedMode(0x0A, "Serial", PinUsage.Uart);
 
         /// <summary>
         /// The pin can be set to input-pullup.
         /// </summary>
-        InputPullup = (0x0B),
+        public static SupportedMode InputPullup = new SupportedMode(0x0B, "Input Pullup", PinUsage.Gpio);
 
         //// Unofficial extensions (only supported by special firmware)
 
         /// <summary>
         /// The pin can be used for SPI transfer (Clock, MOSI, MISO and default CS pin)
         /// For most Arduinos, MOSI=11, MISO=12 and Clock = 13. The default CS pin is 10.
         /// </summary>
-        Spi = 0x0C,
+        public static SupportedMode Spi = new SupportedMode(0x0C, "SPI", PinUsage.Spi);
 
         /// <summary>
         /// HC-SR04
         /// </summary>
-        Sonar = 0x0D,
+        public static SupportedMode Sonar = new SupportedMode(0x0D, "HC-SR04");
 
         /// <summary>
         /// Arduino tone library
         /// </summary>
-        Tone = 0x0E,
+        public static SupportedMode Tone = new SupportedMode(0x0E, "Tone");
 
         /// <summary>
         /// DHT type sensors
         /// </summary>
-        Dht = 0x0F,
+        public static SupportedMode Dht = new SupportedMode(0x0F, "DHT");
+
+        /// <summary>
+        /// Frequency measurement
+        /// </summary>
+        public static SupportedMode Frequency = new SupportedMode(0x10, "Frequency");
+
+        /// <summary>
+        /// Declares a new pin mode
+        /// </summary>
+        /// <param name="value">The pin mode value</param>
+        /// <param name="name">The user-readable name for the mode</param>
+        /// <param name="pinUsage">Pin usage for this mode, if applicable</param>
+        public SupportedMode(byte value, string name, PinUsage pinUsage)
+        {
+            Value = value;
+            Name = name;
+            PinMode = pinUsage;
+        }
+
+        /// <summary>
+        /// Declares a new pin mode
+        /// </summary>
+        /// <param name="value">The pin mode value</param>
+        /// <param name="name">The user-readable name for the mode</param>
+        public SupportedMode(byte value, string name)
+        {
+            Value = value;
+            Name = name;
+            PinMode = PinUsage.Unknown;
+        }
+
+        /// <summary>
+        /// The value for the pin mode
+        /// </summary>
+        public byte Value
+        {
+            get;
+        }
+
+        /// <summary>
+        /// The name of the pin mode
+        /// </summary>
+        public string Name
+        {
+            get;
+        }
+
+        /// <summary>
+        /// The <see cref="PinUsage"/> for this internal mode, if applicable
+        /// </summary>
+        public PinUsage PinMode
```

Comment:
I guess the naming here is a bit confusing, why isn't this called `PinUsage` or `Usage`?